### PR TITLE
Global Registry function for node metadata management

### PR DESCRIPTION
**Pull request by kurt-hectic**
* Introduction of a Global Registry (GR) function to manage metadata needed by GB to discover registered WIS nodes and for WIS nodes to obtain source IP addresses of GB and GC 
* Procedures for WIS nodes to let GB and GC connect 
* Procedures for WIS, GC and GB to manage relevant metadata in the GR

### CODE REVIEW

File: guide/sections/global-services.adoc [hunk 1]
```diff
@@ -117,3 +117,12 @@ TODO: to be completed
 Procedures pertinent to the Global Monitor will be described here once validated through testing during the WIS2 pilot phase.
 
 TODO: to be completed
+
+=== Global Registry ===
+
+* The roles of the Global Registry is to make available the information required for Global Infrastructure and WIS2 nodes to connect with each other.
```
Comment (golfvert) [inline guide/sections/global-services.adoc]: I don't think we need to introduce the term" Global Infrastructure". I'd prefer to keep Global Services. 
We also have to be consistent. Here WIS2 Node is used. In other PR it is WIS Node. So far we used WIS2 Nodes. That would be my preference.

File: guide/sections/global-services.adoc [hunk 2]
```diff
@@ -117,3 +117,12 @@ TODO: to be completed
 Procedures pertinent to the Global Monitor will be described here once validated through testing during the WIS2 pilot phase.
 
 TODO: to be completed
+
+=== Global Registry ===
+
+* The roles of the Global Registry is to make available the information required for Global Infrastructure and WIS2 nodes to connect with each other. 
+* The Global Registry makes available to Global Caches the list of registered WIS2 Nodes. The following metadata is made available for each WIS2 Node: hostname and port(s) of local broker, center_id, list of implemented protocols (MQTT, MQTTs, web-socket).
```
Comment (golfvert) [inline guide/sections/global-services.adoc]: The Global Caches are not on the "front" of the connectivity. GBs are.
So, we could have "The Global Registry makes available to Global Services the list of registered WIS2 Nodes"

File: guide/sections/global-services.adoc [hunk 3]
```diff
@@ -117,3 +117,12 @@ TODO: to be completed
...
+* The Global Registry makes available to WIS2 nodes the list of IP addresses used by Global Brokers and Global Caches to subscribe to and download data from WIS2 nodes.
+* The Global Registry provides authorized contacts the means to update information about their WIS2 Node or Global Infrastrucure.
```
Comment (golfvert) [inline guide/sections/global-services.adoc]: Same here. Infrastructure. Use Services.

**Review by golfvert — CHANGES_REQUESTED** [review]

File: guide/sections/wis2node.adoc [hunk 4]
```diff
@@ -2,11 +2,17 @@
 
 === Practices and procedures
 
-==== Registration and decommissioning of a WIS node
+==== Registration and decommissioning of a WIS2 node
 
-This section describes the process used to register or remove a WIS node within WIS2. During the initial part of the WIS2 pilot phase, a Member simply needs to notify the WMO Secretariat and primary GISC of the intent to register a new WIS node. The Secretariat and GISC will then assist in the registration. More formal procedures will be developed as the number of WIS nodes increases.
+This section describes the process used to register or remove a WIS2 node within WIS2. To register one or more WIS nodes, the Member responside for the node needs to be designated as a NC or DCPC.
```
Comment (golfvert) [inline guide/sections/wis2node.adoc]: WIS or WIS2. I'd prefer WIS2.
Add at the end by the PR of the country.

File: guide/sections/wis2node.adoc [hunk 5]
```diff
@@ -16,9 +22,11 @@ TODO: to be completed
 
 ==== Connecting with Global Services
 
-This section describes the process by which a WIS node is registered with one or more Global Broker and Global Cache components.
+This section describes the process by which a WIS2 node is registered with one or more Global Broker and Global Cache components.
 
-TODO: to be completed
+A WIS2 node shall allow at least one Global Broker and Global Cache to connect to its local broker and web storage. For this, the WIS2 node shall allow the outgoing IP addresses used by the Global Broker and Global Cache to establish connections to broker and web-storage, and allow the Global Cache to subscribe to the topic corresponding to this WIS2 node using the username and password combination "everyone/everyone".
```
Comment (golfvert) [inline guide/sections/wis2node.adoc]: At least two of each... Redundancy.
No, Global Cache don't subscribe to WIS2 Node. Only to GB. 
No one but GBs are subscribing to WIS2 Nodes, GC and GBs.

Replace by:

A WIS2 node shall allow at least two Global Broker to connect to its local broker and at least two Global Cache to its web storage. For this, the WIS2 node shall allow the outgoing IP addresses used by the Global Broker and Global Cache to establish connections to broker and web-storage, and allow the Global Broker to subscribe to the topic corresponding to this WIS2 node. Username and password used to protect the MQTT subscription will be part of the Global Registry.

File: guide/sections/wis2node.adoc [hunk 6]
```diff
@@ -16,9 +22,11 @@ TODO: to be completed
...
+
+The list of IP addresses used by Global Brokers and Caches can be obtained from the Global Registry. If a WIS2 node implements source IP filtering to grant a Global Broker and Global Cache access, it shall implement changes to its filterting within one month of the changes being published to the Global Registry.
```
Comment (golfvert) [inline guide/sections/wis2node.adoc]: The list of IP addresses used by Global Brokers and Caches can be obtained from the Global Registry. If a WIS2 node implements source IP filtering to grant Global Brokers and Global Caches access, it shall implement changes to its filtering within one month of the changes being published to the Global Registry.

**Review by golfvert — CHANGES_REQUESTED** [review]

**Comment by kurt-hectic** [comment]
@golfvert  and @tomkralidis  . Can this be merged? I believe all Remy's comments are addressed. When merging the Secretariat will make sure consistent terminology is used for nodes and services.

**Comment by tomkralidis** [comment]
@kurt-hectic 

- should #27 be closed?
- we have recently updated the structure of the guide, which will required updates to fix the resulting merge conflicts in this PR

**Comment by tomkralidis** [comment]
@kurt-hectic @golfvert closing this PR and invite a new PR based on the new structure (and to resolve merge conflicts).

**Pull request closed without merging**